pkg/ccl/sqlccl/backup.go

```diff
@@ -588,3 +589,77 @@ func MakeRekeyMVCCKeyValFunc(
 		return f(kv)
 	}
 }
+
+func backupPlanHook(
+	baseCtx context.Context, stmt parser.Statement, cfg *sql.ExecutorConfig,
+) (func() ([]parser.DTuple, error), sql.ResultColumns, error) {
+	backup, ok := stmt.(*parser.Backup)
+	if !ok {
+		return nil, nil, nil
+	}
+	header := sql.ResultColumns{
+		{Name: "basePath", Typ: parser.TypeString},
+		{Name: "startTs", Typ: parser.TypeString},
+		{Name: "endTs", Typ: parser.TypeString},
+		{Name: "dataSize", Typ: parser.TypeInt},
+	}
+	return func() ([]parser.DTuple, error) {
```

Comment:
nit (and yeah, I think I might have written this this way originally): IMO multi-value`return` split across 20 lines is a little hard to see what's happening -- I might opt to bind a local var `f` here and below?